dlp-terraform/ecs/ecs_training_service.tf

```diff
@@ -0,0 +1,212 @@
+resource "aws_ecs_task_definition" "training" {
+  family             = "training"
+  task_role_arn      = aws_iam_role.ecs_task_role.arn
+  execution_role_arn = aws_iam_role.ecs_exec_role.arn
+  network_mode       = "bridge"
+  cpu                = 1024
+  memory             = 4096
+
+  container_definitions = jsonencode([
+    {
+      "name" : "training",
+      "image" : "${aws_ecr_repository.training.repository_url}:latest",
+      "portMappings" : [
+        {
+          "name" : "gunicorn-port",
+          "containerPort" : 8000,
+          "hostPort" : 0,
+          "protocol" : "tcp",
+          "appProtocol" : "http"
+        }
+      ],
+      "essential" : true,
+      "environment" : [],
+      "mountPoints" : [],
+      "volumesFrom" : [],
+      "logConfiguration" : {
+        "logDriver" : "awslogs",
+        "options" : {
+          "awslogs-create-group" : "true",
+          "awslogs-region" : "us-east-1",
+          "awslogs-group" : aws_cloudwatch_log_group.training.name,
+          "awslogs-stream-prefix" : "ecs"
+        }
+      }
+    }
+  ])
+}
+
+# --- ECS Service ---
+resource "aws_ecs_service" "training" {
+  name            = "training"
+  cluster         = aws_ecs_cluster.main.id
+  task_definition = aws_ecs_task_definition.training.arn
+  desired_count   = 2
+
+  capacity_provider_strategy {
+    capacity_provider = aws_ecs_capacity_provider.training.name
+    base              = 1
+    weight            = 100
+  }
+
+  ordered_placement_strategy {
+    type  = "spread"
+    field = "attribute:ecs.availability-zone"
+  }
+
+  lifecycle {
+    ignore_changes = [desired_count]
+  }
+
+  # load_balancer {
+  #   target_group_arn = aws_lb_target_group.app.arn
+  #   container_name = "training"
+  #   container_port = 8000
+  # }
+
+  # depends_on = [aws_lb_target_group.app]
+}
+
+# --- ECS Service Auto Scaling ---
+resource "aws_appautoscaling_target" "training_ecs_target" {
+  service_namespace = "ecs"
+  scalable_dimension = "ecs:service:DesiredCount"
+  resource_id = "service/${aws_ecs_cluster.main.name}/${aws_ecs_service.training.name}"
+  min_capacity = 0
```

Comment:
I thought in one of the discussions, you mentioned that the autoscaling target will keep at minimum one instance running even if min capacity is 0? Am I misremembering here? Sry for the basic questions

Reply:
that is correct, it doesn't truly scale to 0

Reply:
so for the problem noah flagged, we don't have a great way to fix it? @andrewpeng02

Reply:
I don't think so, I'l look into it more when I've set up the training stuff.

Reply:
sounds good :)